libraries/ESP8266WiFi/src/WiFiClientSecureBearSSL.h

```diff
@@ -103,6 +103,11 @@ class WiFiClientSecure : public WiFiClient {
     // Sets the requested buffer size for transmit and receive
     void setBufferSizes(int recv, int xmit);
 
+    // Returns whether MFLN negotiation for the above buffer sizes succeeded (after connection)
```

Comment:
> after connection

I'm a bit concerned about the error proneness of this. If the mfln was negotiated, and the result is bigger than the set buffer size, is there some assert or something that can be added somewhere? Attempting to use a connection in such a state is clearly a programming error.

Reply:
It *may be* an error, but is not guaranteed one.  This went back to the early BearSSL days.  You can set buffers any size you like, and BearSSL will try and negotiate to get the server to fit within your receive (actually, now its min(recv buffer, xmit buffer) and I'm talking w/ @pornin since this seems to be over cautious, but that's another PR)).  BearSSL will never overflow buffers, so it's not a write out-of-bounds problem.    If the server only sends fragments which fit w/in the buffer then even if MFLN does not happen things "just work."

If the server sends a fragment that's larger than the receive buffer, it will be dumped by BSSL since you can't decode partial fragments, and then there will be trouble.

So, even on older TLS 1.2 connections, if you're guaranteed the server will send, say, 1K max, then you can safely set a receive buffer of 2K and not have a problem.

BSSL needs the buffers before connections begin, so it's not possible to resize things after-the-fact.

Reply:
My worry is not about overflows et al, but rather about hitting the bad case when establishing a connection, then having fragments silently getting dumped when tryint to use it, then getting a bunch of issues opened about bssl not working.
So what I'm thinking is whether there is some way to completely disallow the bad case: assert, panic, abort, anything to force the user to notice that the set buffer size is smaller than the negotiated size, and hence whether the connection works or not depends on the mood of the nearest plant.

Reply:
The problem with a heavy hammer solution like that is that many people cannot update their server to one that supports MFLN, but they can guarantee the maximum message size (i.e. they're in control of the content returned..either MQTT or CGI response).

It's also only an issue when the `setBufferSizes()` call is used.  By default, BearSSL's completely safe and will connect and transfer from anything, at any size.

Reply:
Alright, if you think the case is valid for a user, I'll defer to you.